reinforced_lib/rlib.py

```diff
@@ -0,0 +1,337 @@
+        the method automatically initializes the first instance.
+
+        Parameters
+        ----------
+        agent_id : int
+            The identifier of agent instance.
+        args : Tuple
+            Environment observations.
+        update_observations : Union[Dict, Tuple, Any]
+            Observations used when 'no_env_mode' is enabled (must match agents 'update_observation_space').
+        sample_observations : Union[Dict, Tuple, Any]
+            Observations used when 'no_env_mode' is enabled (must match agents 'sample_observation_space').
+        kwargs : Dict
+            Environment observations.
+
+        Returns
+        -------
+        out : Any
+            Action selected by the agent.
+        """
+
+        if not self._agent:
+            raise NoAgentError()
+
+        if not self._no_env_mode and not self._env:
+            raise NoEnvironmentError()
+
+        if len(self._agents_states) == 0:
+            self.init()
+
+        self._agents_keys[agent_id], key = jax.random.split(self._agents_keys[agent_id])
+        state = self._agents_states[agent_id]
+
+        if not self._no_env_mode:
+            update_observations, sample_observations = self._env.transform(*args, **kwargs)
+
+        if isinstance(update_observations, dict):
+            state = self._agent(state, **update_observations)
+        elif isinstance(update_observations, tuple):
+            state = self._agent(state, *update_observations)
+        else:
+            state = self._agent(state, update_observations)
+
+        if isinstance(sample_observations, dict):
+            state, action = self._agent(state, key, **sample_observations)
+        elif isinstance(sample_observations, tuple):
+            state, action = self._agent(state, key, *sample_observations)
+        else:
+            state, action = self._agent(state, key, sample_observations)
+
+        self._agents_states[agent_id] = state
+        return action
+
+    def fit(self, agent_id: int = 0) -> None:
+        """
+        Trains selected agent instance.
+
+        Parameters
+        ----------
+        agent_id : int
+            The identifier of agent instance.
+        """
+
+        if not self._agent:
+            raise NoAgentError()
+
+        if not self._no_env_mode and not self._env:
+            raise NoEnvironmentError()
+
+        if len(self._agents_states) == 0:
+            self.init()
+
+        state = self._agents_states[agent_id]
+
+        # TODO develop agents training
+
+        raise NotImplementedError()
+
+    def save_agent_state(self, agent_id: int = 0, path: str = None) -> None:
+        """
+        Saves selected agent instance to file in X format.
+
+        Parameters
+        ----------
+        agent_id : int
+            The identifier of agent instance.
+        path : str
+            Path to the output file.
+        """
+
+        if not path:
```

Comment:
Tutaj w przyszłości (jak już ogarniemy `fit()`) będzie można wyspecifikować jakąś domyślną `path`

Reply:
Też tak pomyślałem, ale na razie zostawiłem puste, bo to jest do ustalenia.

Reply:
To fajnie